### DISCUSSION TITLE
Load multiple ruleset files

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [stoplightio/spectral#1954] [General]
Hi All,

In our validator Spectral is utilised and we are in the situation where two ruleset files should be loaded.

The use cases:
- the tool has its own rulesets and it is provided as a separate npm package and called `fantastic-rulesets`, if the this npm package is not installed then the validator app fails
- the user can provide his/her own rulesets using a `.spectral.yaml` file, it must `extend` the `fantastic-rulesets` package

Loading the npm package is easy, I created a loader which pulls the rules together in an object and will be handed over Spectral (`setRuleset` method) and call it a day.
But, in the second case I use `migrateRuleset` from `@stoplight/spectral-ruleset-migrator` which, if I assume correctly, ensures some compatibility (which is awesome) and deals with different file formats (also awesome). But, it returns a javascript file, and this is pain in the neck. I either have to use `eval()` to deal with the file or some other black magic.

Is there a recommended way to deal with that I have multiple sources of rulesets and they have to be loaded into Spectral?

Thanks for any help!

**Comment by P0lip** [comment]
As mentioned here https://github.com/stoplightio/spectral/issues/1956, I'm going provide to a util to make it easier to load a YAML/JSON ruleset when using Spectral programmatically.
Hopefully, that reduces the complexity of your code.

**Reply by CornerTakenQuickly** [reply]
Please provide an example showing how to load multiple rulesets in Spectral v6.x. We have been using spectral v5.x for quite some time. Our solution runs the base Spectral OAS rules, plus about 10 custom rulesets at once. We load multiple rulesets using

`swaggerLinter.setRuleset(cust_rules);`

where cust_rules is an array of filepaths.

We are attempting to migrate to Spectral v6.x, and the capability to load multiple rulesets seems to have been removed. The code mentioned in #1956 still seems to allow only a single ruleset to be loaded.
